src/backend/api/meals.js

```diff
@@ -1,15 +1,151 @@
 const express = require("express");
+const { limit } = require("../database");
 const router = express.Router();
 const knex = require("../database");
 
+
```

Comment:
This handler suffers from the same problem as the warmup exercise. It has many `if` steps and here they are also `nested` and each block is fairly big. This is hard to understand and update, and there is a fair amount of duplication that probably isn't needed.

Further, the handler doesn't support the case when all or some of the flags are combined in new ways which is part of the exercise.

The solution is similar to what I pointed out in https://github.com/negaren/hyf-homework/pull/19/files#r826736613. You want to find a way to let the most of your code work in the same way.

In this case, it's less about converting to a shared format but rather to find default values and then adjust the code to act as if these parameters are ALWAYS set. Like this:
```js
router.get("/", async (request, response) => {
  try {
    const { maxPrice, availableReservations, title, createdAfter, limit } = unpackParams(request);
   
    const meals = await knex('meal').select('*');
    const reservations = await knex('reservation').select('*');

    const mealsBelowMaxPrice = meals .filter((meal) => meal.price < maxPrice);
    
    const mealsBigEnoughForGroup = mealsBelowMaxPrice.filter((meal) => {
      const reservationsForMeal = reservations .filter((reservation) => meal.id === reservation.meal_id)
      const totalNumberOfGuests = reservationsForMeal.reduce((acc, reservation) => acc + reservation.number_of_guests, 0);
      return meal.max_reservations > totalNumberOfGuests;
    });

    const mealsForSearch = mealsBigEnoughForGroup.filter((meal) => meal.title.includes(title)); // You could do this in Knex also

   const newMeals = mealsForSearch.filter((meal) => meal.created_date > createdAfter); // You could do this in Knex also

   const topMeals = newMeals.slice(0, limit);
   
    response.json(topMeals);
  } catch (error) {
    console.log(error);
    response.status(500).send('Internal server error');
  }
});

function unpackParams(request) {
  const defaultParams = {
    maxPrice: Number.MAX_VALUE,
    availableReservations: Number.MAX_VALUE,
    title: '',
    createdAfter: ,
    limit: Number.MAX_VALUE
  };
  
  const params = Object.assign({}, defaultParams, request.query); // Merge defaults with request parameters

  // Logic that validates that all values make sense, using if, throw if something is not valid.

  return params;
}
```

Like this, we've moved the `if` logic into the function `unpackParams` and the rest of the code always follows the same flow which makes it easier to read. To understand what will happen, we can log the converted params and from those values we should know exactly what the code will do.